From 5c591aeb2a194a9554b0cf0bd3959d8c18fa5129 Mon Sep 17 00:00:00 2001
From: Pavel Emelyanov <xemul@openvz.org>
Date: Wed, 14 Jan 2009 18:23:02 +0300
Subject: [PATCH] bridge: don't leak master device on brctl addif

If we add a second ethernet device to bridge the former one leaks.

http://bugzilla.openvz.org/show_bug.cgi?id=1145

Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 net/bridge/br_if.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/bridge/br_if.c b/net/bridge/br_if.c
index 3dac8fc..4588ddc 100644
--- a/net/bridge/br_if.c
+++ b/net/bridge/br_if.c
@@ -406,7 +406,7 @@ int br_add_if(struct net_bridge *br, struct net_device *dev)
 	if ((dev->flags & IFF_UP) && netif_carrier_ok(dev) &&
 	    (br->dev->flags & IFF_UP))
 		br_stp_enable_port(p);
-	if (!(dev->features & NETIF_F_VIRTUAL)) {
+	if (!(dev->features & NETIF_F_VIRTUAL) && !br->master_dev) {
 		dev_hold(dev);
 		br->master_dev = dev;
 	}
-- 
1.6.0.6

